charts/openfaas-cloud/templates/of-builder-dep.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.ofBuilder.name }}
+  labels:
+    app.kubernetes.io/name: openfaas-cloud
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/component: {{ .Values.ofBuilder.name }}
+spec:
+  replicas: {{ .Values.ofBuilder.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ .Values.ofBuilder.name }}
+  template:
+    metadata:
+      annotations:
+        prometheus.io.scrape: {{ .Values.global.prometheusScrape | quote }}
+      labels:
+        app: {{ .Values.ofBuilder.name }}
+    spec:
+      volumes:
+        - name: registry-secret
+          secret:
+            secretName: registry-secret
+        - name: payload-secret
+          secret:
+            secretName: payload-secret
+{{- if .Values.global.enableECR }}
+        - name: aws-ecr-credentials
+          secret:
+            defaultMode: 420
+            secretName: aws-ecr-credentials
+{{- end }}
+      containers:
+        - name: {{ .Values.ofBuilder.name }}
+          image: {{ .Values.ofBuilder.image }}:{{ .Values.ofBuilder.tag }}
+          imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8080
+            initialDelaySeconds: 2
+            periodSeconds: 10
+            timeoutSeconds: 2
+          env:
+            - name: enable_lchown
+              value: "true"
+            - name: insecure
+              value: "false"
+            - name: buildkit_url
+              value: "tcp://127.0.0.1:1234"
+            - name: "disable_hmac"
+              value: "false"
+          ports:
+            - containerPort: 8080
+              protocol: TCP
+          volumeMounts:
+            - name: registry-secret
+              readOnly: true
+              mountPath: "/home/app/.docker/"
+            - name: payload-secret
+              readOnly: true
+              mountPath: "/var/openfaas/secrets/"
+{{- if .Values.global.enableECR }}
+            - name: aws-ecr-credentials
+              mountPath: /home/app/.aws/
+              readOnly: true
+{{- end }}
+        - name: of-buildkit
+          args: ["--addr", "tcp://0.0.0.0:1234"]
+          image: {{ .Values.ofBuilder.buildKitImage }}:{{ .Values.ofBuilder.buildKitTag }}
+          imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+          ports:
+            - containerPort: 1234
+              protocol: TCP
+          securityContext:
+            privileged: true
```

Comment:
There is a way to run as `privileged=false` and non-root which it would be good to have as an option

> Using Rootless mode (*.rootless.yaml) is recommended because Rootless mode image is executed as non-root user (UID 1000) and doesn't need securityContext.privileged.

https://github.com/moby/buildkit/tree/master/examples/kubernetes

Reply:
It might be something to look at in this PR or with a follow-up issue and PR, this chart will certainly make it easier to experiment.

Reply:
Its easy to add that in now, so ill just pop it in. 👍